pkg/predicate/predicate.go

```diff
@@ -270,3 +270,16 @@ func (o or) Generic(e event.GenericEvent) bool {
 	}
 	return false
 }
+
+// EventReconcileFilter takes in a LabelSelector as a parameter and returns predicate functions. The labels of
+// events that are contained as parameters in these predicate functions are compared against the instantiated
+// Selector object and reconciliation of the event will only take place if the labels match.
+func EventReconcileFilter(s metav1.LabelSelector) (Funcs, error) {
```

Comment:
Please rename to `LabelSelectorPredicate`, no one will find this by its current name. Same goes for the PR title.
Also, the PR has merge conflicts and needs to be squashed so there is only one commit.

Reply:
Will do.

**Quick question:**
The conflict was caused because a lot of the event struct's fields have changed (since I forked this repo), now taking in a controllerutil object. The events require a `controllerutil.Object` type (interface), which I believe can be implemented as `metav1.Object`. However, the type that supports labels as metadata is the `metav1.ObjectMeta`. How would I initialize a `controllerutil.Object` -typed object that contains labels in it? That would help me write some tests.

Thanks!

Reply:
All kube objects that are not lists fullfil the `controllerutil.Object` interface, for example `*corev1.Pod`